### DISCUSSION TITLE
Should we change the used logging library?

### DISCUSSION THREAD

**Discussion by kwitsch** [0xERR0R/blocky#1275] [Ideas]
Since there are multiple libraries for structured logging and `logrus` (which we currently use) falls behind in most benchmarks we should reevaluate it.

Should we change it?
If yes, to which?

**Comment by ThinkChaos** [comment]
Go now has the [slog](https://pkg.go.dev/log/slog) module in the standard library, probably worth a look but I've never used it personally.

**Reply by kwitsch** [reply]
I came across [zap](https://github.com/uber-go/zap) which also has an `slog` integration.
The comparison on there repo got me thinking...😅

**Comment by 0xERR0R** [comment]
I used zap, zerolog and slog in my other projects. All of them are not compatible with logrus, so each log statement must be changed. Advantage of slog: there is no need to include some dependencies and I would personalty prefer slog (I think golang team will optimize it in the future). There is also fast growing comunity with additionional projects like github.com/lmittmann/tint

Regarding benchmarks: the question is, even if logrus is slower, does it impact the overall performance of blocky? We do have very good test coverage and a lot of tests (also e2e) but no automatic benchmarking/profiling. Maybe we should do something in this direction? so we can identify potential bottlenecks and slow implementations.
